### DISCUSSION TITLE
Should unverified user get the auth token

### DISCUSSION THREAD

**Discussion by laszlo1337** [pocketbase/pocketbase#2630] [Q&A] [answered]
I have noticed that even though the user account is unverified, when calling `authWithPassowrd` the auth token is still issued. Is that by design? if yes, what is the rationale? 

`authWithPassword()`:
```
{
  record: Record {
    avatar: '',
    collectionId: '_pb_users_auth_',
    collectionName: 'users',
    created: '2023-06-01 13:03:44.018Z',
    email: 'xxx@gmail.com',
    emailVisibility: false,
    id: 'chyn2iu2hm1q2ea',
    name: '',
    updated: '2023-06-01 13:03:44.018Z',
    username: 'users72291',
    verified: false,
    expand: {}
  },
  token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF8iLCJleHAiOjE2ODY5MTU2NjMsImlkIjoiY2h5bjJpdTJobTFxMmVhIiwidHlwZSI6ImF1dGhSZWNvcmQifQ.0lyk6Tey2cwSXJe-mh_BqhskXf4
e5S8VfOLC6ellMnE'
}

```

**Comment by ganigeorgiev** [comment, marked as answer]
It is intentional. You may want to allow access to some part of the app even if the user is not verified. You can disallow unverified user access in the API rules by checking `@request.auth.verified = true`.

Eventually in the future we may add a global auth collection option that will disallow authentication if the user is not verified.

**Reply by laszlo1337** [reply]
Where should I specify that rule? `@request.auth.verified = true`
I immediately assumed it would be somewhere here:
![image](https://github.com/pocketbase/pocketbase/assets/24696743/044fc8d4-f555-4177-b910-469de1c2118a)

but none of those seem right?

**Reply by ganigeorgiev** [reply]
> but none of those seem right?

I'm not sure that I follow. The `@request.auth.verified = true` should be added to the collections that you want to restrict access to.

**Reply by laszlo1337** [reply]
Ideally I'd want a failed response from `authWithPassword` if user is unverified. Is that possible?

**Reply by ganigeorgiev** [reply]
There is no API rule for that. At the moment, the only way to return an error would be to use PocketBase as framework and attach for example to the [`OnRecordAuthRequest` hook](https://pocketbase.io/docs/event-hooks/#record-auth-api-hooks).
